superset/connectors/druid/models.py

```diff
@@ -799,11 +777,54 @@ def recursive_get_fields(_conf):
                     post_aggs[metric_name] = HyperUniqueCardinality(
                         mconf.get('name')
                     )
-                else:
+                elif mconf.get('type') == 'arithmetic':
                     post_aggs[metric_name] = Postaggregator(
                         mconf.get('fn', "/"),
                         mconf.get('fields', []),
                         mconf.get('name', ''))
+                else:
+                    post_aggs[metric_name] = Postaggregator(
+                        None,
+                        None,
+                        mconf.get('name', ''))
+                    post_aggs[metric_name].post_aggregator = mconf
+        return all_metrics, post_aggs
+
+    def run_query(  # noqa / druid
```

Comment:
I'm not sure why git is getting the diffing all wrong here, things are not lining up. Makes it hard to see what has changed in this section. Maybe moving `_metrics_and_post_aggs` higher / lower in the class definition might help git diffing it right?